src/main/java/com/github/kongchen/swagger/docgen/SwaggerAnnotationIntrospector2.java

```diff
@@ -0,0 +1,27 @@
+package com.github.kongchen.swagger.docgen;
+
+import com.fasterxml.jackson.databind.PropertyName;
+import com.fasterxml.jackson.databind.introspect.AnnotatedClass;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.jackson.SwaggerAnnotationIntrospector;
+
+/**
+ * Extends SwaggerAnnotationIntrospector with {@link #findRootName(AnnotatedClass)} implementation. See
+ * https://github.com/swagger-api/swagger-core/issues/2104
+ * 
+ * @author Tomasz Juchniewicz
+ *
+ */
+public class SwaggerAnnotationIntrospector2 extends SwaggerAnnotationIntrospector {
```

Comment:
This class needs a better name than just adding "2". Please make it descriptive of functionality.